### PR TITLE
Replace Flask-RESTPlus with Flask-RESTX

### PR DESCRIPTION
**Pull request by bluemellophone**
## Pull Request Overview

- This PR replaces the dependency `Flask-RESTPlus` with [`Flask-RESTX`](https://flask-restx.readthedocs.io/en/latest/index.html#) due to [reasons](https://flask-restx.readthedocs.io/en/latest/index.html#why-did-we-fork).  See here for discussion: https://github.com/noirbizarre/flask-restplus/issues/593
- Fixes the `patch` and `delete` REST resources for `Submissions` with the use of the `return_not_found` flag on the `@api.resolve_object_by_model` decorator.
- Fixed the broken tests for Submissions and adding functionality to actually test patching of Submissions
- Fixed OAuth2 session key cleanup from testing in `AutoAuthFlaskClient`
- Rename all TODO comment instances of `# @todo` to standardized `# TODO:`  (Note @hwindsor)
- Re-linted some files not in the PR

---

**Pull Request Checklist**
- [x] Ensure that the PR is properly formatted
- [x] Ensure that the PR is properly rebased
  - The PR is rebased on `develop` (commit: `a6001e3`)
- [x] Ensure that the PR uses a consolidated database migration
  - _No migration_
- [x] Ensure that the PR is properly tested
- [x] Ensure that the PR is properly sanitized
- [x] Ensure that the PR is properly reviewed
 - @mmulich you flagged this dependency initially as being EOL, please review this PR to make sure we area ready to migrate to an active community-supported fork.

### CODE REVIEW

File: flask_restx_patched/swagger.py [hunk 1]
```diff
@@ -21,3 +21,20 @@ def parameters_for(self, doc):
...
+            assert params is not None
+
+            for name, param in params.items():
+                get_func = getattr(param, 'get', None)
+                if get_func is None:
+
+                    def _patched_get(value, default):
+                        return param.context.get(value, default)
+
+                    param.get = _patched_get
```
Comment (mmulich) [inline flask_restx_patched/swagger.py:37]: What's this patching about?

File: tests/modules/submissions/resources/test_permissions.py [hunk 2]
```diff
@@ -104,20 +103,43 @@ def test_create_patch_submission(flask_app_client, regular_user, readonly_user,
...
             )
         assert readonly_delete_response.status_code == 403
+
         with flask_app_client.login(regular_user, auth_scopes=('submissions:write',)):
             regular_delete_response = flask_app_client.delete(
                 '/api/v1/submissions/%s' % submission_guid
             )
-        # @todo, this shouldn't give an internal error but it's not clear why it is
-        assert regular_delete_response.status_code == 500
+        assert regular_delete_response.status_code == 204
```
Comment (mmulich) [inline tests/modules/submissions/resources/test_permissions.py:142]: Nice work fixing this test!

**Review by mmulich — DISMISSED** [review]
Looks good to me 👍🏼

File: tests/modules/submissions/resources/test_permissions.py (hunk 2 above)
Comment (bluemellophone) [reply tests/modules/submissions/resources/test_permissions.py:142]: 👍

File: flask_restx_patched/swagger.py (hunk 1 above)
Comment (bluemellophone) [reply flask_restx_patched/swagger.py:37]: The error was tracked down to [here](https://github.com/python-restx/flask-restx/blob/master/flask_restx/swagger.py#L350).  There is a call a few lines above on line 343 that I am using to hack together support for a `get` method on the parameter object.  I originally had overridden each of the Parameter types we use from `flask_marshmallow.base_fields` with a Mix-in for just that method, but found this to be a more extensible solution. 

I wanted this PR to be a minimal drop-in replacement with just the minimum changed.  I propose that a refactoring of the codebase in relation to updating (and hopefully slimming down) `flask_restx_patched` will come in the future.  This implementation of `expected_params` is to get the ball rolling with a working migration.  I think this [line](https://github.com/WildMeOrg/houston/blob/master/flask_restplus_patched/namespace.py#L92) will offer some evidence on how to eliminate the need for this patched function.

**Review by bluemellophone — COMMENTED** [review]
I've self-reviewed these changes, LGTM

File: flask_restx_patched/swagger.py (hunk 1 above)
Comment (bluemellophone) [reply flask_restx_patched/swagger.py:37]: By the way, I'm not entirely convinced this patch is the correct path to take.  There seems to be upstream state being set by Swagger and the implementation of `extract_resource_doc` between `Flask-RESTPlus` and `Flask-RESTX` are identical in their expectations of a `get` function.  After tearing into this for a week, and about three alternatives, this attempts to squash the downstream state issue and gets us moving forward until a more wholistic audit of what is actually needed in `flask_restx_patched` can be performed.

**Review by colinwkingen — APPROVED** [review]
LGTM, glad it was drop-in. Thanks for doing this.

**Pull request merged**